src/distilabel/llm/litellm.py

```diff
@@ -0,0 +1,183 @@
+                text so far. Defaults to `None`.
+            frequency_penalty: It is used to penalize new tokens based on their frequency in the
+                text so far. Defaults to `None`.
+            logit_bias: Used to modify the probability of specific tokens appearing in the
+                completion. Defaults to `None`.
+            user: A unique identifier representing your end-user. This can help the LLM provider
+                to monitor and detect abuse. Defaults to `None`.
+            metadata: Pass in additional metadata to tag your completion calls - eg. prompt
+                version, details, etc. Defaults to `None`.
+            api_base: Base URL for the API. Defaults to `None`.
+            api_version: API version. Defaults to `None`.
+            api_key: API key. Defaults to `None`.
+            model_list: List of api base, version, keys. Defaults to `None`.
+            mock_response: If provided, return a mock completion response for testing or debugging
+                purposes. Defaults to `None`.
+            force_timeout: The maximum execution time in seconds for the completion request.
+                Defaults to `600`.
+            custom_llm_provider: Used for Non-OpenAI LLMs, Example usage for bedrock, set(iterable)
+                model="amazon.titan-tg1-large" and custom_llm_provider="bedrock". Defaults to
+                `None`.
+
+        Returns:
+            A list of lists of strings containing the generated responses for each input.
+        """
+        import litellm
+
+        async def _call_aclient_untill_n_choices():
+            choices = []
+            while len(choices) < num_generations:
+                completion = await self._aclient(
+                    model=self.model,
+                    messages=input,
+                    n=num_generations,
+                    functions=functions,
+                    function_call=function_call,
+                    temperature=temperature,
+                    top_p=top_p,
+                    stream=False,
+                    stop=stop,
+                    max_tokens=max_tokens,
+                    presence_penalty=presence_penalty,
+                    frequency_penalty=frequency_penalty,
+                    logit_bias=logit_bias,
+                    user=user,
+                    metadata=metadata,
+                    api_base=api_base,
+                    api_version=api_version,
+                    api_key=api_key,
+                    model_list=model_list,
+                    mock_response=mock_response,
+                    force_timeout=force_timeout,
+                    custom_llm_provider=custom_llm_provider,
+                )
+                choices.extend(completion.choices)
+            return choices
+
+        try:
+            litellm.drop_params = False
+            choices = await _call_aclient_untill_n_choices()
```

Comment:
Typo `untill`->`until`

```suggestion
            choices = await _call_aclient_until_n_choices()
```